[BUG] CMake: add_library cannot create ALIAS target "Boost::headers" [...]

## Issue description

On my system, on 0b9acc40, I get this error:
```
  add_library cannot create ALIAS target "Boost::headers" because target
  "Boost::boost" is IMPORTED.
```

## Reproducible example code

~Not sure if this is fully reproducible, but this is what I have on Ubuntu 18.04:~

Here's the workflow from scratch using `ubuntu:18.04` in docker:

```
$ docker image ls ubuntu:18.04 
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
ubuntu              18.04               56def654ec22        3 weeks ago         63.2MB
$ docker run -it --rm ubuntu:18.04
# apt update
# apt install -y build-essential cmake git libboost1.65-dev python3-dev python3-pytest
# git clone https://github.com/pybind/pybind11
# cd pybind11/
# git log -n 1 --oneline --no-decorate
0b9acc40 fix: chapters in PDF again (#2606)
# mkdir build && cd build
# cmake ..
-- The CXX compiler identification is GNU 7.5.0
...
-- pybind11 v2.6.0 
-- CMake 3.10.2
-- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9") 
...
-- Boost version: 1.65.1
CMake Error at tests/CMakeLists.txt:231 (add_library):
  add_library cannot create ALIAS target "Boost::headers" because target
  "Boost::boost" is IMPORTED.
```

If I disable this branch (e.g. `if(FALSE)` instead), then it at least configures well:
https://github.com/pybind/pybind11/blob/0b9acc4009afdda427b5f151a67d500216707dc1/tests/CMakeLists.txt#L229

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -226,23 +226,12 @@ find_package(Boost 1.56)
 
 if(Boost_FOUND)
   if(NOT TARGET Boost::headers)
-    set(use_alias FALSE)
+    add_library(Boost::headers IMPORTED INTERFACE)
     if(TARGET Boost::boost)
-      get_property(
-        is_imported
-        TARGET Boost::boost
-        PROPERTY IMPORTED)
-      if(NOT ${is_imported})
-        set(use_alias TRUE)
-      endif()
-    endif()
-
-    if(${use_alias})
       # Classic FindBoost
-      add_library(Boost::headers ALIAS Boost::boost)
+      set_property(TARGET Boost::boost PROPERTY INTERFACE_LINK_LIBRARIES Boost::boost)
     else()
       # Very old FindBoost, or newer Boost than CMake in older CMakes
-      add_library(Boost::headers IMPORTED INTERFACE)
       set_property(TARGET Boost::headers PROPERTY INTERFACE_INCLUDE_DIRECTORIES
                                                   ${Boost_INCLUDE_DIRS})
     endif()

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -226,7 +226,18 @@ find_package(Boost 1.56)
 
 if(Boost_FOUND)
   if(NOT TARGET Boost::headers)
+    set(use_alias FALSE)
     if(TARGET Boost::boost)
+      get_property(
+        is_imported
+        TARGET Boost::boost
+        PROPERTY IMPORTED)
+      if(NOT ${is_imported})
+        set(use_alias TRUE)
+      endif()
+    endif()
+
+    if(${use_alias})
       # Classic FindBoost
       add_library(Boost::headers ALIAS Boost::boost)
     else()

